ftd/src/executor/styles.rs

```diff
@@ -823,22 +1032,365 @@ impl Background {
+            ),
+            true,
+        )
+    }
+
+    pub fn background_size_pattern() -> (String, bool) {
+        (
+            format!(
+                indoc::indoc! {"
+                let bg = {{0}};
+                if (typeof bg === 'object' && \"size\" in bg) {{
+                    let sz = bg.size;
+                    if (typeof sz === 'object' && \"x\" in sz && \"y\" in sz) {{
+                        sz.x + \" \" + sz.y
+                    }}
+                    else {{
+                        sz
+                    }}
+                }} else {{
+                    \"{ignore_key}\"
+                }}
+            "},
+                ignore_key = ftd::interpreter2::FTD_IGNORE_KEY
+            ),
+            true,
+        )
+    }
+
+    pub fn background_position_pattern() -> (String, bool) {
+        (
+            format!(
+                indoc::indoc! {"
+                let bg = {{0}};
+                if (typeof bg === 'object' && \"position\" in bg) {{
+                    let pos = bg.position;
+                    if (typeof pos === 'object' && \"x\" in pos && \"y\" in pos) {{
+                        pos.x + \" \" + pos.y
+                    }}
+                    else {{
+                        pos.replace(\"-\", \" \")
+                    }}
+                }} else {{
+                    \"{ignore_key}\"
+                }}
+            "},
+                ignore_key = ftd::interpreter2::FTD_IGNORE_KEY
+            ),
+            true,
+        )
+    }
+}
+
+#[derive(serde::Deserialize, Debug, PartialEq, Clone, serde::Serialize)]
+pub enum BackgroundRepeat {
+    Repeat,
+    RepeatX,
+    RepeatY,
+    NoRepeat,
+    Space,
+    Round,
+}
+
+impl BackgroundRepeat {
+    fn from_optional_value(
+        value: ftd::interpreter2::PropertyValue,
+        doc: &ftd::executor::TDoc,
+        line_number: usize,
+    ) -> ftd::executor::Result<Option<BackgroundRepeat>> {
+        match value.value(doc.name, line_number)? {
+            ftd::interpreter2::Value::Optional { data, .. } if data.is_none() => Ok(None),
+            _ => Ok(Some(BackgroundRepeat::from_value(value, doc, line_number)?)),
+        }
+    }
+
+    fn from_value(
+        value: ftd::interpreter2::PropertyValue,
+        doc: &ftd::executor::TDoc,
+        line_number: usize,
+    ) -> ftd::executor::Result<BackgroundRepeat> {
+        let binding = value.resolve(&doc.itdoc(), line_number)?;
+        let value = binding.get_or_type(doc.name, line_number)?;
+        let value = (value.1.to_owned(), value.2.to_owned());
+        BackgroundRepeat::from_values(value, doc, line_number)
+    }
+
+    fn from_values(
+        or_type_value: (String, ftd::interpreter2::PropertyValue),
+        doc: &ftd::executor::TDoc,
+        line_number: usize,
+    ) -> ftd::executor::Result<Self> {
+        match or_type_value.0.as_str() {
+            ftd::interpreter2::FTD_BACKGROUND_REPEAT_BOTH_REPEAT => Ok(BackgroundRepeat::Repeat),
```

Comment:
```suggestion
            ftd::interpreter2::FTD_BACKGROUND_REPEAT_BOTH_REPEAT => Ok(ftd::executor::BackgroundRepeat::Repeat),
```

Do this for the rest.